relay-server/src/services/project_cache.rs

```diff
@@ -907,40 +909,33 @@ impl ProjectCacheBroker {
     fn handle_periodic_unspool(&mut self) {
         self.buffer_unspool_handle.reset();
 
+        // If we don't yet have the global config, we will defer dequeuing until we do.
+        if let GlobalConfigStatus::Pending = self.global_config {
+            self.buffer_unspool_backoff.reset();
+            self.schedule_unspool();
+            return;
+        }
         // If there is nothing spooled, schedule the next check a little bit later.
-        if self.index.is_empty() {
+        // And do *not* attempt to unspool if the assigned permits over low watermark.
+        if self.index.is_empty() && !self.buffer_guard.is_below_low_watermark() {
```

Comment:
shouldn't this be or `||` ? Like isn't this two different conditions that both would imply we should skip spooling?

Reply:
yeah,  it should be `or` here. Thanks!